src/cart/src/cart/crt_swim.c

```diff
@@ -114,6 +119,41 @@ static void crt_swim_srv_cb(crt_rpc_t *rpc_req)
 		rpc_req->cr_opc, rpc_swim_input->upds.ca_count,
 		self_id, rpc_swim_input->src);
 
+	rpc_priv = container_of(rpc_req, struct crt_rpc_priv, crp_pub);
+	delay = (hlc - rpc_priv->crp_req_hdr.cch_hlc) / NSEC_PER_MSEC;
+	csm->csm_msg_count++;
+
+	if (csm->csm_hlc) {
+		csm->csm_avg_delay = (csm->csm_avg_delay + delay) / 2;
+
+		D_TRACE_DEBUG(DB_TRACE, rpc_req,
+			      "%lu. delay: %lu ms, avg: %lu ms, max: %lu ms\n",
+			      csm->csm_msg_count, delay, csm->csm_avg_delay,
+			      max_delay);
+
+		if (delay > max_delay) {
+			swim_net_glitch_update(ctx, delay);
```

Comment:
swim_net_glitch_update(ctx, delay - max_delay)?

Reply:
You right, but I'd like to be more aggressive and see exact number of delay in error message from this function.